### PR TITLE
CI-faq-q2

### PR DESCRIPTION
**Pull request by kartikeya2609**
CI-faq-q2

Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: CI-faq-q2

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [X] Successful preview build.

### CODE REVIEW

File: kb/continuous-integration/continuous-integration-faqs.md [hunk 1]
```diff
@@ -2349,6 +2349,48 @@ curl -X POST -i -H "Authorization: Bearer AUTH_TOKEN" https://<DRONE_SERVER_FQDN
...
+
+```
+This ensures the **Save Cache** step runs only if it's the first instance (iteration 0) of the parallel stages and if the current stage execution is successful.
+
+
+### How does caching work when cloning pipelines with Save/Restore Cache steps?
+When the user clones a pipeline with caching steps, the cache keys generated by the cloned pipeline use the original pipeline's cache key as a prefix. For instance, if the original cache key is `some-cache-key`, the cloned pipeline might use `some-cache-key2`. This prevents accidental interference between cache keys of the original and cloned pipelines.
+
+### How can user ensure correct cache handling when cloning pipelines with Save Cache to GCS?
+Enable separators (/) for GCS cache keys by setting PLUGIN_ENABLE_SEPARATOR: true in your pipeline's stage variables.
```
Comment (dewan-ahmed) [inline kb/continuous-integration/continuous-integration-faqs.md:2373]:
```suggestion
Enable separators (/) for GCS cache keys by setting `PLUGIN_ENABLE_SEPARATOR: true` in your pipeline's stage variables.
```

File: kb/continuous-integration/continuous-integration-faqs.md [hunk 2]
```diff
@@ -2349,6 +2349,48 @@ curl -X POST -i -H "Authorization: Bearer AUTH_TOKEN" https://<DRONE_SERVER_FQDN
...
+
+
+### How can user resolve S3 bucket permission issue in save cache in S3 step?
+If aws connector is configured with the Cross account role ARN, so user need to configure the PLUGIN_USER_ROLE_ARN stage variable as suggested in the below doc:
+https://developer.harness.io/docs/continuous-integration/use-ci/build-and-upload-artifacts/build-and-push/build-and-push-to-ecr-step-settings/#aws-connector
```
Comment (dewan-ahmed) [inline kb/continuous-integration/continuous-integration-faqs.md:2378]:
```suggestion
If the AWS connector is configured with the cross-account role ARN, the user needs to configure the `PLUGIN_USER_ROLE_ARN` stage variable as suggested in [this doc](../../docs/continuous-integration/use-ci/build-and-upload-artifacts/build-and-push/build-and-push-to-ecr-step-settings/#aws-connector)
```

File: kb/continuous-integration/continuous-integration-faqs.md [hunk 3]
```diff
@@ -2349,6 +2349,48 @@ curl -X POST -i -H "Authorization: Bearer AUTH_TOKEN" https://<DRONE_SERVER_FQDN
...
+
+
+### Can user import test results from a generic 'mvn clean deploy' Run step into the test Dashboard? Or must use the Unit Test step to display them on that tab?
+Yes, you can use the Run step to run your maven command, and upload the test results. There is a field for this in the run step (like in the run tests step). Please see
+https://developer.harness.io/docs/continuous-delivery/x-platform-cd-features/cd-steps/containerized-steps/run-step/#report-paths for more information.
```
Comment (dewan-ahmed) [inline kb/continuous-integration/continuous-integration-faqs.md:2383]:
```suggestion
Yes, you can use the Run step to run your maven command and upload the test results. There is a field for this in the run step (like in the run tests step). Please check [the documentation](../../docs/continuous-delivery/x-platform-cd-features/cd-steps/containerized-steps/run-step/#report-paths) for more information.
```

**Review by dewan-ahmed — COMMENTED** [review]
Hi @kartikeya2609 thanks for your contribution. Please address the review comments; mainly to use relative links in docs. Thanks.

**Comment by bot2-harness** [comment]
This pull request has been automatically closed due to 30 days of inactivity. If you believe this was done in error, please feel free to reopen the PR or reach out to the BT team for further assistance. Thank you for your contributions!

**Pull request closed without merging**